charts/nri-kube-events/templates/_helpers_compatibility.tpl

```diff
@@ -3,8 +3,330 @@ Returns a dictionary with legacy runAsUser config.
+{{- define "nri-kube-events.compatibility.old.integration.registry" -}}
+    {{- if .Values.image -}}
+        {{- if .Values.image.kubeEvents -}}
+            {{- if .Values.image.kubeEvents.registry -}}
+                {{ .Values.image.kubeEvents.registry }}
+            {{- end -}}
+        {{- end -}}
+    {{- end -}}
+{{- end -}}
+
+{{- define "nri-kube-events.compatibility.old.integration.repository" -}}
+    {{- if .Values.image -}}
+        {{- if .Values.image.kubeEvents -}}
+            {{- if .Values.image.kubeEvents.repository -}}
+                {{ .Values.image.kubeEvents.repository }}
+            {{- end -}}
+        {{- end -}}
+    {{- end -}}
+{{- end -}}
+
+{{- define "nri-kube-events.compatibility.old.integration.tag" -}}
+    {{- if .Values.image -}}
+        {{- if .Values.image.kubeEvents -}}
+            {{- if .Values.image.kubeEvents.tag -}}
+                {{ .Values.image.kubeEvents.tag }}
+            {{- end -}}
+        {{- end -}}
+    {{- end -}}
+{{- end -}}
+
+{{- define "nri-kube-events.compatibility.old.integration.pullPolicy" -}}
+    {{- if .Values.image -}}
+        {{- if .Values.image.kubeEvents -}}
+            {{- if .Values.image.kubeEvents.pullPolicy -}}
+                {{ .Values.image.kubeEvents.pullPolicy }}
+            {{- end -}}
+        {{- end -}}
+    {{- end -}}
+{{- end -}}
+
+{{- define "nri-kube-events.compatibility.old.agent.registry" -}}
+    {{- if .Values.image -}}
+        {{- if .Values.image.infraAgent -}}
+            {{- if .Values.image.infraAgent.registry -}}
+                {{ .Values.image.infraAgent.registry }}
+            {{- end -}}
+        {{- end -}}
+    {{- end -}}
+{{- end -}}
+
+{{- define "nri-kube-events.compatibility.old.agent.repository" -}}
+    {{- if .Values.image -}}
+        {{- if .Values.image.infraAgent -}}
+            {{- if .Values.image.infraAgent.repository -}}
+                {{ .Values.image.infraAgent.repository }}
+            {{- end -}}
+        {{- end -}}
+    {{- end -}}
+{{- end -}}
+
+{{- define "nri-kube-events.compatibility.old.agent.tag" -}}
+    {{- if .Values.image -}}
+        {{- if .Values.image.infraAgent -}}
+            {{- if .Values.image.infraAgent.tag -}}
+                {{ .Values.image.infraAgent.tag }}
+            {{- end -}}
+        {{- end -}}
+    {{- end -}}
+{{- end -}}
+
+{{- define "nri-kube-events.compatibility.old.agent.pullPolicy" -}}
+    {{- if .Values.image -}}
+        {{- if .Values.image.infraAgent -}}
+            {{- if .Values.image.infraAgent.pullPolicy -}}
+                {{ .Values.image.infraAgent.pullPolicy }}
+            {{- end -}}
+        {{- end -}}
+    {{- end -}}
+{{- end -}}
+
+{{- /* Functions to fetch integration and agent image configurations from the new .Values.images */ -}}
+{{- define "nri-kube-events.compatibility.new.integration.registry" -}}
+    {{- if .Values.images -}}
+        {{- if .Values.images.integration -}}
```

Comment:
could we assume all `values.images` to be not-nil since they are specified in the values?

Reply:
Now that you said that, we can get rid of all the `new`  helpers.

Reply:
Fixed in [#bd7e656](https://github.com/newrelic/nri-kube-events/pull/117/commits/bd7e65639f8c8c9fcfb6dfed470483c40260388b)